tests/storages_tests/test_with_server.py

```diff
@@ -211,3 +235,88 @@ def test_get_best_trial(direction: StudyDirection, values: Sequence[float]) -> N
     )
     expected_value = max(values) if direction == StudyDirection.MAXIMIZE else min(values)
     assert study.best_value == expected_value
+
+
+def test_set_and_get_study_user_attrs_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+
+    # Test setting value.
+    for key, value in FLOAT_ATTRS.items():
+        storage.set_study_user_attr(study_id, key, value)
+        assert is_equal_floats(storage.get_study_user_attrs(study_id)[key], value)
+
+
+def test_set_and_get_study_system_attrs_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+
+    # Test setting value.
+    for key, value in FLOAT_ATTRS.items():
+        storage.set_study_system_attr(study_id, key, value)
+        assert is_equal_floats(storage.get_study_system_attrs(study_id)[key], value)
+
+
+def test_set_trial_state_values_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+    for value in FLOAT_ATTRS.values():
+        if math.isnan(value):  # NOTE: Optuna does not accept `nan` as `value`.
+            continue
+        trial_id = storage.create_new_trial(study_id)
+        storage.set_trial_state_values(trial_id, state=TrialState.COMPLETE, values=(value,))
+        set_value = storage.get_trial(trial_id).value
+        assert set_value is not None
+        assert is_equal_floats(set_value, value)
+
+
+def test_set_trial_param_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+    trial_id = storage.create_new_trial(study_id)
+
+    for key, value in FLOAT_ATTRS.items():
+        float_distribution = FloatDistribution(low=value, high=value)
+        categorical_distribution = CategoricalDistribution(choices=(value,))
+        for distribution in (float_distribution, categorical_distribution):
+            # NOTE: suggest_float does not generate NaN, and MySQL cannot handle infinities.
```

Comment:
Same as I mentioned in `test_storages.py`.
```suggestion
            # NOTE: `suggest_float` does not generate `nan` or `±inf`, and MySQL cannot handle `±inf`.
```